src/settings.rs

```diff
@@ -16,13 +16,21 @@ pub struct Settings {
 
 #[derive(Debug, Clone)]
 pub struct BackendSettings {
-    pub shell: String,
+    pub program: String,
+    pub args: Vec<String>,
+}
+
+impl BackendSettings {
+    pub fn new(program: String, args: Vec<String>) -> Self {
```

Comment:
I think that constructor function is not needed for config structs because we can expect a lot of parameters, so we will need to provide either builder pattern or `new` method with too many arguments.